sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -900,12 +862,22 @@ int gcoap_resp_init(coap_pkt_t *pdu, uint8_t *buf, size_t len, unsigned code)
     }
     coap_hdr_set_code(pdu->hdr, code);
 
-    /* Reserve some space between the header and payload to write options later */
-    pdu->payload      = buf + coap_get_total_hdr_len(pdu) + GCOAP_RESP_OPTIONS_BUF;
-    /* Payload length really zero at this point, but we set this to the available
-     * length in the buffer. Allows us to reconstruct buffer length later. */
-    pdu->payload_len  = len - (pdu->payload - buf);
-    pdu->content_type = COAP_FORMAT_NONE;
+    unsigned header_len  = sizeof(coap_hdr_t) + coap_get_token_len(pdu);
```

Comment:
This looks like a `coap_get_total_hdr_len(pdu);`

Reply:
I agree. It looks like I don't even need the header_len variable itself. I'll take a closer look.